service/storage_service.cc

```diff
@@ -745,12 +771,13 @@ void storage_service::handle_cdc_generation(std::optional<db_clock::time_point>
 
     bool using_this_gen = false;
     try {
-        using_this_gen = do_handle_cdc_generation(*ts);
-    } catch(...) {
-        cdc_log.warn("Could not retrieve CDC streams with timestamp {} on gossip event: {}."
-                " Will retry in the background.", *ts, std::current_exception());
+        using_this_gen = do_handle_cdc_generation_intercept_nonfatal_errors(*ts);
+    } catch (cdc_generation_handling_nonfatal_exception& e) {
+        cdc_log.warn(could_not_retrieve_msg_template, ts, e.what(), "retrying in the background");
         async_handle_cdc_generation(*ts);
```

Comment:
Nit: no need to remove this `return`, but it doesn't matter I guess.

Reply:
done